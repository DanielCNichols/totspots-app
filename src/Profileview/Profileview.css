.profileview {
  width: 90%;
  margin: 100px auto;
}

@media screen and (min-width: 450px) {
  .profileview {
    width: 70%;
  }
}

@media screen and (min-width: 800px) {
  .profileview {
    width:35%;
  }
}
