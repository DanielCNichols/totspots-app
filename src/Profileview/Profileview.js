import React from 'react';
import venuesContext from '../VenuesContext'
import ApiService from '../services/api-service';
import FavoritesList from '../FavoritesList/Favoriteslist';
import ReviewsList from '../ReviewsList/ReviewsList'
import './Profileview.css'

export default class ProfileView extends React.Component {
  static contextType = venuesContext;

  componentDidMount() {
    this.context.clearError()
    ApiService.getProfile()
    .then(profile => {
      this.context.setProfile(profile)
    })
    .catch(this.context.setError)
  }



  render() {
    const {profile, error} = this.context
    return (
      <section className="profileview">
        <header>
          <h3>Your profile</h3>
        </header>
        <div role='alert'>
          {error && <p className='error'>{error}</p>}
        </div>
        <p>Name: {profile.first_name} {profile.last_name}</p>
        <p>Email: {profile.email}</p>
        <p>UserName: {profile.username}</p>
        <div>
          <FavoritesList></FavoritesList>
        </div>
        <div>
          <ReviewsList></ReviewsList>
        </div>
      </section>
    );
  }
}
