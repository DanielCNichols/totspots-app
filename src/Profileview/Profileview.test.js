import React from 'react';
import ReactDOM from 'react-dom'
import Profileview from './Profileview.js'
import {MemoryRouter} from 'react-router-dom'
import renderer from 'react-test-renderer'

const profile = {
  username: "nkirsop1",
  first_name: "Nevil",
  last_name: "Kirsop",
  city: null,
  state: null,
  email: "nkirsop1@goodreads.com"
}


it('renders Profileview without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
  <MemoryRouter>
    <Profileview />
  </MemoryRouter>, div);
  ReactDOM.unmountComponentAtNode(div)
})

it('renders Profileview as expected', () => {
  const tree = renderer
  .create(
  <MemoryRouter>
    <Profileview name="Profileview" profile= {profile}/>
  </MemoryRouter>
  )
  .toJSON();
  expect(tree).toMatchSnapshot();
})


it('renders Profileview as expected with no profile', () => {
  const tree = renderer
  .create(
  <MemoryRouter>
    <Profileview name="Profileview" profile= {[]}/>
  </MemoryRouter>
  )
  .toJSON();
  expect(tree).toMatchSnapshot();
})
