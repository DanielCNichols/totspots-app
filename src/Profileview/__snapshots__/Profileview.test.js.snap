// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Profileview as expected 1`] = `
<section
  className="profileview"
>
  <header>
    <h3>
      Your profile
    </h3>
  </header>
  <div
    role="alert"
  />
  <p>
    Name: 
     
  </p>
  <p>
    Email: 
  </p>
  <p>
    UserName: 
  </p>
  <div>
    <div>
      <div>
        <p>
          You haven't saved any venues!
        </p>
      </div>
    </div>
  </div>
  <div>
    <div>
      <div>
        <p>
          Looks like you haven't posted any reviews... yet!
        </p>
      </div>
    </div>
  </div>
</section>
`;

exports[`renders Profileview as expected with no profile 1`] = `
<section
  className="profileview"
>
  <header>
    <h3>
      Your profile
    </h3>
  </header>
  <div
    role="alert"
  />
  <p>
    Name: 
     
  </p>
  <p>
    Email: 
  </p>
  <p>
    UserName: 
  </p>
  <div>
    <div>
      <div>
        <p>
          You haven't saved any venues!
        </p>
      </div>
    </div>
  </div>
  <div>
    <div>
      <div>
        <p>
          Looks like you haven't posted any reviews... yet!
        </p>
      </div>
    </div>
  </div>
</section>
`;
